### PR TITLE
feat: various usability & stabiliy improvements for interop

### PR DESCRIPTION
**Pull request by laurentsenta**
- [x] Introduce `split` operation in composition templates
- [x] Add a `load_resource` operation in composition templates
  - This lets a user define configuration in a file (like a list of instances) and use them during composition templating.
- [x] drop arbitrary build timeout (we have a task timeout)
- [x] drop instance count requirements when we can deduce it
- [x] provide composition templating during `build` (it was only applied during `run` before)
- [x] Add a concurrency option to build artifacts one by one
  - When working on rust interop tests, I noticed we lose the benefits of docker caching & explodes storage requirements because we start every build at the same time.

### CODE REVIEW

**Comment by laurentsenta** [comment]
Marking for review,
Note each commit is one feature, I think having a single PR will save us time, but I can extract 1 PR per commit if needed.

File: pkg/cmd/fixtures/templates/with-resource-complex.toml
Comment (brdji) [inline pkg/cmd/fixtures/templates/with-resource-complex.toml:20]: This looks very unwieldy and prone to typos. I would prefer the much cleaner option of using something like `go_build_image=golang:1.18-buster` in the `resource.toml` file itself.

File: pkg/build/docker_generic.go
Comment (brdji) [inline pkg/build/docker_generic.go:50]: Do we have an integration test that checks the task timeout?

Comment (laurentsenta) [reply pkg/build/docker_generic.go:50]: Not really, here we're dropping a 30 minutes timeout: it's a rather trivial change, and the only integration test we could write here is to wait for 30 minutes in the CI. The benefits / cost ratio would be very low.

File: pkg/cmd/fixtures/templates/with-resource-complex.toml
Comment (laurentsenta) [reply pkg/cmd/fixtures/templates/with-resource-complex.toml:20]: So the discussion is whether the benefit of reusing a single variable (go_version) in multiple locations is worth the complexity it adds to the template definition. That might be fine in some cases, and not fine in other cases.

But that's something a Testground User might worry about.

We're wearing a Testground Maintainer hat here: we're not using or maintaining this composition file, we're only testing the templating feature with a few fixtures that looks like real tests. I believe the test would be as good or worst if we dropped this part because it would cover a different, smaller, use case.

Comment (brdji) [reply pkg/cmd/fixtures/templates/with-resource-complex.toml:20]: Understood. Feel free to close this.

**Review by brdji — APPROVED** [review]

**Pull request merged**
